rules/windows/command_and_control_certutil_network_connection.toml

```diff
@@ -26,22 +26,32 @@ in order to take the next steps in a compromised environment.
+
 #### Possible investigation steps
 
 - Investigate the script execution chain (parent process tree) for unknown processes. Examine their executable files for
 prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
 - Investigate other alerts associated with the user/host during the past 48 hours.
 - Investigate if the downloaded file was executed.
 - Determine the context in which `certutil.exe` and the file were run.
-- Retrieve the downloaded file and determine if it is malicious:
-  - Use a private sandboxed malware analysis system to perform analysis.
-    - Observe and collect information about the following activities:
-      - Attempts to contact external domains and addresses.
-      - File and registry access, modification, and creation activities.
-      - Service creation and launch activities.
-      - Scheduled task creation.
-  - Use the PowerShell Get-FileHash cmdlet to get the files' SHA-256 hash values.
-    - Search for the existence and reputation of the hashes in resources like VirusTotal, Hybrid-Analysis, CISCO Talos, Any.run, etc.
+- Examine the host for derived artifacts that indicates suspicious activities:
+  - Analyze the downloaded file using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by
+      filtering by the process' `process.entity_id`.
+      - Examine the DNS cache for suspicious or anomalous entries.
+        - !{osquery{"query":"SELECT * FROM dns_cache", "label":"Osquery - Retrieve DNS Cache"}}
+    - Use the Elastic Defend registry events to examine registry keys accessed, modified, or created by the related
+    processes in the process tree.
+    - Examine the host services for suspicious or anomalous entries.
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services","label":"Osquery - Retrieve All Services"}}
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services WHERE NOT (user_account LIKE \"%LocalSystem\" OR user_account LIKE \"%LocalService\" OR user_account LIKE \"%NetworkService\" OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
+      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != \"trusted\"","label":"Osquery - Retrieve Service Unsigned Executables with Virustotal Link"}}
```

Comment:
Do you have screenshots of these rendering and producing the desired results?

Reply:
![image](https://user-images.githubusercontent.com/26856693/202539273-f81f362d-24be-42f2-9c67-52ccd5fd3109.png)
![image](https://user-images.githubusercontent.com/26856693/202539775-c366dc1b-cdee-49a3-af23-7f792c84b596.png)
![image](https://user-images.githubusercontent.com/26856693/202539837-94a892be-f52e-497c-86e9-8efd0645c10f.png)
![image](https://user-images.githubusercontent.com/26856693/202540280-4ecf9f91-e504-40ad-9aaf-30e4a2b00c5d.png)